app/views/unattended/provisioning_templates/snippet/kickstart_rhsm.erb

```diff
@@ -0,0 +1,30 @@
+<%#
+kind: snippet
+name: kickstart_rhsm
+model: ProvisioningTemplate
+snippet: true
+description: |
+  Generate Anaconda RHSM configuration for Red Hat OS registration
+test_on:
+- rhel9_dhcp
+-%>
+<%
+  subman_keys = host_param('kt_activation_keys') || host_param('activation_key')
+  subman_org = host_param('subscription_manager_org')
+  subman_registration = host_param_true?('subscription_manager') || subman_keys
+  subman_hostname = " --server-hostname #{host_param('subscription_manager_server_hostname')}" if host_param('subscription_manager_server_hostname')
+  subman_rhsm_baseurl = " --rhsm-baseurl #{host_param('subscription_manager_rhsm_baseurl')}" if host_param('subscription_manager_rhsm_baseurl')
```

Comment:
@M4rtinK What's the difference between `server-hostname` and `rhsm-baseurl`?

Reply:
AFAIK `rhsm-baseurl` is related to the Red Hat CDN if the regular Red Hat hosted subscription infra is used or to Satellites equivalent of it.

In the past we have set both `server-hostname` and `rhsm-baseurl` to use the staging version of the Red Hat subscription infra (so staging hosted Candlepin and staging CDN) during testing.

But for registration to Satellite & using its repositories only the `server-hostname` needs to be set, as the `katello-rhsm-consumer` provisioning script takes care of setting the `rhsm-baseurl`.

I think this might actually be the bit that generates the part of the script that does that ?

https://github.com/Katello/katello/blob/c5f8e9532e61f9e5c7ec648decf81660628dd4ba/app/helpers/katello/content_source_helper.rb#L39

Reply:
Thanks @M4rtinK!

> But for registration to Satellite & using its repositories only the server-hostname needs to be set, as the katello-rhsm-consumer provisioning script takes care of setting the rhsm-baseurl.

@ekohl So we can go only with the `server-hostname` parameter, but TBH I believe we still could have both there, just in case.
The question is - do we wan't to load the values from host_params or from template macros?

Reply:
I forgot to link it back, but in https://github.com/pykickstart/pykickstart/pull/425#discussion_r963038800 I asked for clarification if `rhsm-baseurl` is even used when `server-hostname` is specified. If it is, I feel like specifying it is a good thing but it may not make sense.

Reply:
In [redhat_register.erb](https://github.com/theforeman/foreman/blob/develop/app/views/unattended/provisioning_templates/snippet/redhat_register.erb) we don't have support for configuration of `rhsm-baseurl` nor `server-hostname`, we can only configure `proxy_hostname` (if it is set by host_param).

Since this PR is mainly to fix broken RHEL 9 provisioning, and we need to get it into the 6.12, how about to do it now without the support of `server-hostname` & `rhsm-baseurl`?

We can do it later in follow up PR implementing these parameters for anaconda's `rhsm` (RHEL 9) and also in `redhat_register.erb` for RHEL 7 & 8.

@ekohl what do you think?

Reply:
> Since this PR is mainly to fix broken RHEL 9 provisioning, and we need to get it into the 6.12, how about to do it now without the support of `server-hostname` & `rhsm-baseurl`?

AFAIK that's incorrect since it means you register to the CDN rather than a Satellite. You must use `server-hostname` for RHEL 9.

Today it's done via the `subscription_manager_certpkg_url` here:
https://github.com/theforeman/foreman/blob/b6b47f05278434354f6dac043c2e13f71a5975d4/app/views/unattended/provisioning_templates/snippet/redhat_register.erb#L168-L172

That would need to be replaced. I'd be fine with only doing so via the host macros and not having host parameters for it. Then you must select a correct content source when creating a host, but IMHO it's a misconfiguration if you don't.

Reply:
Okay, so we can keep as it is now, for rhel there are going to be two new parameters `subscription_manager_server_hostname` and `subscription_manager_rhsm_baseurl` and that's it, right?

There haven't been any updates in https://github.com/pykickstart/pykickstart/pull/425#discussion_r963038800, and 6.12 is pretty close so we need to push it forward.

Reply:
I'd also be OK without any host parameters, only support what's defined via the content source if that helps you push it forward faster.